app/controllers/authentications_controller.rb

```diff
@@ -1,56 +1,64 @@
+# allows a session to be identified and potentially invalidated by the owning user or administrators.
 class AuthenticationsController < ApplicationController
   before_action :authorize_access_request!, only: [:destroy]
 
   # login
   def create
     user = User.find_by!(email: params[:email])
 
-    if user.authenticate(params[:password])
-
-      payload = {user_id: user.id}
-      session = JWTSessions::Session.new(payload: payload, refresh_by_access_allowed: true)
-      tokens = session.login
+    raise JWTSessions::Errors::Unauthorized, 'Authentication failed' unless user.authenticate(params[:password])
 
-      response.set_cookie(
-          JWTSessions.access_cookie,
-          value: tokens[:access],
-          httponly: true,
-          secure: Rails.env.production?
-      )
+    # got here, means authentication was sucessful
 
-      # record session information in the database for administrative use
-      browser = Browser.new(request.headers['User-Agent'])
-      Session.create!(
-          browser: browser.name,
-          browser_version: browser.full_version,
-          device: browser.device.name,
-          expiring_at: Time.now, # TODO: must pull this from the refresh token
-          ip_address: request.remote_ip,
-          platform: browser.platform.name,
-          platform_version: browser.platform.version,
-          ruid: session.payload['ruid'],
-          user: user
-      )
+    payload = { user_id: user.id }
+    session = JWTSessions::Session.new(payload: payload, refresh_by_access_allowed: true)
+    tokens = session.login
 
-      render json: {csrf: tokens[:csrf]}, status: :created
+    response.set_cookie(
+      JWTSessions.access_cookie,
+      value: tokens[:access],
+      httponly: true,
+      secure: Rails.env.production?
+    )
 
-    else
+    # record session information in the database for administrative use
+    browser = Browser.new(request.headers['User-Agent'])
+    Session.create!(
+      browser: browser.name,
+      browser_version: browser.full_version,
+      device: browser.device.name,
+      expiring_at: tokens[:refresh_expires_at],
+      ip_address: request.remote_ip,
+      platform: browser.platform.name,
+      platform_version: browser.platform.version,
+      ruid: session.payload['ruid'],
+      user: user
+    )
 
-      raise JWTSessions::Errors::Unauthorized, 'Authentication failed'
-
-    end
+    render json: { csrf: tokens[:csrf] }, status: :created
   end
 
   # logout
   def destroy
-    session = JWTSessions::Session.new(payload: payload)
-    session.flush_by_access_payload
+    jwt_session = JWTSessions::Session.new(payload: payload)
+    jwt_session.flush_by_access_payload
 
     response.delete_cookie JWTSessions.access_cookie
-    # TODO update SessionsUser
 
-    render json: {data: {}, meta: {}}, status: :no_content
+    Session.find_by(ruid: payload['ruid']).update!(invalidated: true, invalidated_by: current_user)
```

Comment:
Here's the other place where we need to look up the `Session` information and make sure it is invalidated.